src/governance-classes.cpp

```diff
@@ -0,0 +1,671 @@
+                        govobj->fExpired = true;
+                    }
+                }
+                remove = true;
+                break;
+            case SEEN_OBJECT_IS_VALID:
+                {
+                    // Rough approximation: 30 days per month * 576 blocks per day
+                    static const int nMonthlyBlocks = 30*576;
+                    int nTriggerBlock = superblock->GetBlockStart();
+                    int nExpirationBlock = nTriggerBlock + nMonthlyBlocks;
+                    if(governance.GetCachedBlockHeight() > nExpirationBlock)  {
+                        remove = true;
+                        CGovernanceObject* govobj = superblock->GetGovernanceObject();
+                        if(govobj)  {
+                            govobj->fExpired = true;
+                        }
+                    }
+                }
+                break;
+            default:
+                break;
+            }
+        }
+        
+        if(remove)  {
+            DBG( 
+                string strdata = "NULL";
+                CGovernanceObject* govobj = superblock->GetGovernanceObject();
+                if(govobj)  {
+                    strdata = govobj->GetDataAsString();
+                }
+                cout << "CGovernanceTriggerManager::CleanAndRemove: Removing object: " 
+                     << strdata
+                     << endl;
+               );
+            mapTrigger.erase(it++);
+        }
+        else  {
+            ++it;
+        }
+    }
+
+    DBG( cout << "CGovernanceTriggerManager::CleanAndRemove: End" << endl; );
+}
+
+/**
+*   Get Active Triggers
+*
+*   - Look through triggers and scan for active ones
+*   - Return the triggers in a list
+*/
+
+std::vector<CSuperblock_sptr> CGovernanceTriggerManager::GetActiveTriggers()
+{
+    AssertLockHeld(governance.cs);
+    std::vector<CSuperblock_sptr> vecResults;
+
+    DBG( cout << "GetActiveTriggers: mapTrigger.size() = " << mapTrigger.size() << endl; );
+
+    // LOOK AT THESE OBJECTS AND COMPILE A VALID LIST OF TRIGGERS
+    trigger_m_it it = mapTrigger.begin();
+    while(it != mapTrigger.end()) {
+
+        CGovernanceObject* pObj = governance.FindGovernanceObject((*it).first);
+
+        if(pObj) {
+            DBG( cout << "GetActiveTriggers: pObj->GetDataAsString() = " << pObj->GetDataAsString() << endl; );
+            vecResults.push_back(it->second);
+        }
+        ++it;
+    }
+
+    DBG( cout << "GetActiveTriggers: vecResults.size() = " << vecResults.size() << endl; );
+
+    return vecResults;
+}
+
+/**
+*   Is Superblock Triggered
+*
+*   - Does this block have a non-executed and actived trigger?
+*/
+
+bool CSuperblockManager::IsSuperblockTriggered(int nBlockHeight)
+{
+    if (!IsValidSuperblockHeight(nBlockHeight)) {
+        return false;
+    }
+
+    LOCK(governance.cs);
+    // GET ALL ACTIVE TRIGGERS
+    std::vector<CSuperblock_sptr> vecTriggers = triggerman.GetActiveTriggers();
+    //int nYesCount = 0;
```

Comment:
I'd prefer to leave it for now.  Uncommenting line 310 is the patch I apply for dev testing.  (Although next time I'll use a small payment amount to avoid knocking out any more testnet nodes).